web/src/inbox_ui.ts

```diff
@@ -597,12 +638,35 @@ function rerender_stream_inbox_header_if_needed(
     }
 }
 
+function get_channel_folder_header_id(folder_id: number): string {
+    if (folder_id === OTHER_CHANNELS_FOLDER_ID) {
+        return OTHER_CHANNEL_HEADER_ID;
+    } else if (folder_id === PINNED_CHANNEL_FOLDER_ID) {
+        return PINNED_CHANNEL_HEADER_ID;
+    }
+    return CHANNEL_FOLDER_HEADER_ID_PREFIX + folder_id;
+}
+
+function rerender_channel_folder_header_if_needed(
+    old_folder_data: ChannelFolderContext,
+    new_folder_data: ChannelFolderContext,
+): void {
+    for (const property of channel_folder_context_properties) {
+        if (new_folder_data[property] !== old_folder_data[property]) {
```

Comment:
This loop is a bit weird; I feel like a single `if (_.any(...))` clause comparing the fields  ... or just an `_.isEqual` would be more natural?

Reply:
I can open a separate PR for this. Currently it follows the pattern of other `rerender_foo_if_needed` functions.